packages/@ember/engine/index.ts

```diff
@@ -2,18 +2,20 @@ export { getEngineParent, setEngineParent } from './lib/engine-parent';
 
 import { canInvoke } from '@ember/-internals/utils';
 import Controller from '@ember/controller';
-import { Namespace, RegistryProxyMixin } from '@ember/-internals/runtime';
+import { Namespace } from '@ember/-internals/runtime';
 import { Registry } from '@ember/-internals/container';
 import DAG from 'dag-map';
 import { assert } from '@ember/debug';
 import { get, set } from '@ember/-internals/metal';
-import EngineInstance from '@ember/engine/instance';
+import EngineInstance, { EngineInstanceOptions } from '@ember/engine/instance';
 import { RoutingService } from '@ember/-internals/routing';
 import { ContainerDebugAdapter } from '@ember/-internals/extension-support';
 import { ComponentLookup } from '@ember/-internals/views';
 import { setupEngineRegistry } from '@ember/-internals/glimmer';
+import RegistryProxyMixin from '@ember/-internals/runtime/lib/mixins/registry_proxy';
+import { ResolverClass } from '@ember/-internals/container/lib/registry';
 
-function props(obj) {
+function props(obj: any) {
```

Comment:
1. If we need to keep this, [`object` should be safe][1].

2. As far as I can tell, `for...in` has *exactly* the same semantics as `Object.keys()`:

    > `Object.keys()` returns an array whose elements are strings corresponding to the enumerable properties found directly upon `object`. The ordering of the properties is the same as that given by looping over the properties of the object manually.

    —[MDN `Object.keys()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys)

    > The for...in statement iterates over all [enumerable properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties) of an object that are keyed by strings (ignoring ones keyed by [Symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)), including inherited enumerable properties.

    —[MDN `for...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)

[1]: https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABABwE52QZwBRwEYBWAXIvgQKbQCUiA3gFCKIA25UK6y5qs5mJmKKhhgA5gG0AuogC8iKQG56jRMDipE2VuwDW5AJ6IRpQjQZMmaDN16YAdMhCYAFtj36qSpgF9lTVGwgqEhWXDwwfEq+QA

Net, I think we can just drop this in favor of `Object.keys()`.

Reply:
Making this change causes tests to fail... I haven't figured out why so I just left it as-is.